OnlineShop.Services.Catalog.Api/Controllers/CatalogController.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using OnlineShop.Services.Catalog.Application.Models.Dto;
+using OnlineShop.Services.Catalog.Application.Services.Interfaces;
+
+namespace OnlineShop.Services.Catalog.Api.Controllers
+{
+    [Route("api/products")]
+    [ApiController]
+    public class CatalogController : ControllerBase
+    {
+        private readonly ICatalogService _catalogService;
+
+        public CatalogController(ICatalogService catalogService)
+        {
+            _catalogService = catalogService;
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetProductsAsync()
+        {
+            var response = await _catalogService.GetAllProductsAsync();
+
+            return Ok(response);
+        }
+
+        [HttpGet("{id:length(24)}")]
+        public async Task<IActionResult> GetProductAsync(string id)
```

Comment:
It is better to specify, in which way you will get id. This can be achieved using attributes